narwhal/primary/src/tests/certificate_waiter_tests.rs

```diff
@@ -427,7 +442,14 @@ async fn fetch_certificates_basic() {
     certs.push(cert);
     // Add cert with incorrect digest.
     let mut cert = certificates[num_written].clone();
-    cert.header.id = Default::default();
+    // This is a bit tedious to craft
+    let cert_header = unsafe { std::mem::transmute::<Header, BadHeader>(cert.header) };
```

Comment:
Do we know if this conversion is sound?

Reply:
Hhhm what if you used the `HeaderBuilder` to set the id? You could introduce a custom method, like the [with_payload_batch](https://github.com/MystenLabs/sui/blob/cfd8464b00c4c505fe4fb97463cc6d5ac81723b9/narwhal/types/src/primary.rs#L170) to achieve that. Then you wouldn't need the `BadHeader` struct as well